feat: updated website landing page and integrated blogs

Hetarth02 opened the pull request:
View the demo here: https://hetarth02.github.io/JuliaHealthBlog/

**1st PR** Towards #24.

- added home page
- added blogs page
~- added connect with us page~ (will be in follow-up PR)

Hetarth02 committed:
- WIP | updated website
- added home page
- added blogs page
- added connect with us page

TheCedarPrince commented:
Wow wow wow -- I can only give quick thoughts at the moment; I'll CC some folks to see if they can help finish getting this PR set-up. @Jay-sanjay, @jaredSchwartz, and/or @divital-coder, could any of you help with doing a PR review of Hetarth's work here?

Some quick thoughts:

1. Is this all still quarto?
2. I think we should migrate blog posts in a separate PR -- would be too much for this PR
3. The landing page looks FANTASTIC. I love it personally.
4. I think adding in the other organizations should be in another PR
5. Highlighting packages should also be in another PR
6. I really like the FAQs but am curious if we should have that be its own dedicated page or somehow link to that section on the landing page. What do you think?
7. Do you know if there is a better Dark Mode toggle? I am not quite a fan of the look of the current dark mode -- but that's just me.

Thanks so so much @Hetarth02 ! Excited to see how we can get this merged soon! If you can answer and help with these questions I pointed out above, I think that would be great and make getting this merged in much faster.

Cheers!

~ tcp :deciduous_tree:

jakubMitura14 commented:
Indeed looks nice ! I think some detailed instructions how to add data and not break it would be usefull also

Hetarth02 commented:
@jakubMitura14

> I think some detailed instructions how to add data and not break it would be usefull also

Do you mean, you want to experiment with code?

If so, it's still in the early stages. Right now I just statically put all the data in quarto files(look at `index.qmd`).

Have to come up with some way to feed it using a `json` or `julia` script later.

jakubMitura14 commented:
Ah OK 👍

Hetarth02 commented:
> 1. Is this all still quarto?

Yup, everything is quarto and just a little bit of HTML, CSS and bootstrap 5(this is in-built in quarto)
Unless and until necessary I don't plan to deviate from quarto(tools provided by quarto)...

> 2. I think we should migrate blog posts in a separate PR -- would be too much for this PR

I personally am in support of dividing up a big PR into smaller manageable parts, but this time I think at the very least we would need to add Landing page and Blogs together. The main reason is there might be some downtime between doing subsequent PRs and while other things like related orgs page, meeting notes, etc. can wait Blogs and Landing page cannot.

> 3. The landing page looks FANTASTIC. I love it personally.

Thanks, more updates and fine-tuning to come. Look forward to it.😄 

> 4. I think adding in the other organizations should be in another PR
> 5. Highlighting packages should also be in another PR

Yeah, fair enough!

> 6. I really like the FAQs but am curious if we should have that be its own dedicated page or somehow link to that section on the landing page. What do you think?

Depends on the number of FAQs we want to put, if it's like 3 or 5 in total it doesn't make sense to have a page dedicated to it.

> 7. Do you know if there is a better Dark Mode toggle? I am not quite a fan of the look of the current dark mode -- but that's just me.

Might need to research a bit on this.

TheCedarPrince commented:
> Unless and until necessary I don't plan to deviate from quarto(tools provided by quarto)...

Sounds great and works well here.

> The main reason is there might be some downtime between doing subsequent PRs and while other things like related orgs page, meeting notes, etc. can wait Blogs and Landing page cannot.

That's reasonable. In that case, go on ahead with migrating the blog posts over as needed is my thought.

> Depends on the number of FAQs we want to put, if it's like 3 or 5 in total it doesn't make sense to have a page dedicated to it.

Sounds good -- we can worry about a dedicated FAQ page later too as you suggested. 

> Might need to research a bit on this.

Yea, no worries at all on Dark Mode. We can definitely wait on that for a separate PR. I think trying to chase this right now would delay this PR. 

Hope that helps Hetarth!

Hetarth02 committed:
- remove dark mode for now
- added images in package and org cards
- made home page more responsive for all devices
- added all blogs from old website

Hetarth02 commented:
Referring https://github.com/JuliaHealth/juliahealth.github.io/issues/131 for future ref.

Hetarth02 committed:
- minor typo fixes

Hetarth02 committed:
- add rendered files

Hetarth02 committed:
- added links to cards

Hetarth02 committed:
- commented out wip routes and pages
- added some packages and orgs
- updated blogs route

Hetarth02 committed:
- updated title for blogs page

Hetarth02 committed:
- updated title for blogs page

cncastillo commented:
Amazing! It is looking very nice. 

Two issues:
- Currently the text that says "And many more..." doesn't point to anything.
- I would suggest to remove the `width=10em` and `card-img-top` from the card images's style. I think it looks better:

![image](https://github.com/user-attachments/assets/46a0849f-b81a-40e3-bffb-75420f9e431d)

Hetarth02 commented:
> Amazing! It is looking very nice.

Thank you so much!😄

> Currently the text that says "And many more..." doesn't point to anything.

Yeah, the packages and orgs page will be in a follow up PR.

> remove the `width=10em` and `card-img-top` from the card images's style.

I need to check if works with different number of cards, responsiveness, etc. If it's unnecessary will remove it.

Thanks for the feedback!👍
